### DISCUSSION TITLE
Whats best-practice for response mapping in Akka .NET?

### DISCUSSION THREAD

**Discussion by swimmesberger** [akkadotnet/akka.net#5540] [Q&A]
I would like to know how response mapping can/should be implemented in Akka .NET without using await and blocking the mailbox.
If for example I have following hierarchy:
Actor A ---RequestB--->    Actor B  ---RequestC----> Actor C
Actor C ---ResponseC---> Actor B ---ResponseB---> Actor A

What is the best way to map the responses from ResponseC to ResponseB? 
I can only imagine passing all information that is on the stack to put into RequestC and add to the protocol that ResponseC passed the information put into RequestC back so I can add Receive<ResponseC> were I can Tell to the original Actor A requester based on the passed context information in the response but that feels unergonomic.

Java Akka has a concept of message adapters* that seems to fit my use case so I wondered if there is something in Akka .NET that I'm missing?

\* https://doc.akka.io/docs/akka/current/typed/interaction-patterns.html#adapted-response

**Comment by eaba** [comment]
@swimmesberger sorry for the delay in responding. We will get back to you on this!
